src/components/dependencies/RescoringModal.js

```diff
@@ -434,6 +435,9 @@ const RescoringFilterOption = ({
     // nothing to do
     if (selectedSet.isSubsetOf(optionsSet)) return
 
+    // options not yet fetched
+    if (isLoading) return
```

Comment:
I'd expect this early-exit at the very beginning of this `useEffect` since it's not of great value to do the checks with the `optionsSet` if options are not yet fetched anyways.